python/tvm/relay/op/nn/_nn.py

```diff
@@ -125,20 +125,22 @@ def schedule_conv2d(attrs, outs, target):
     groups = attrs.groups
     layout = attrs.data_layout
     kernel_layout = attrs.kernel_layout
+    channels = attrs.channels #TODO should be input channels
     with target:
         if groups == 1 and layout == "NCHW":
             return topi.generic.schedule_conv2d_nchw(outs)
         if groups == 1 and layout == "NCHW4c":
             return topi.generic.schedule_conv2d_nchw(outs)
         if groups == 1 and layout == "NHWC":
             return topi.generic.schedule_conv2d_nhwc(outs)
-        if groups != 1:
+        if groups == channels:
             if layout == "NCHW":
                 # TODO(leyuan, merrymercy, Huyuwei): fold depthwise topi into conv2d.
                 return topi.generic.schedule_depthwise_conv2d_nchw(outs)
             if layout == "NHWC" and kernel_layout == "HWOI":
                 return topi.generic.schedule_depthwise_conv2d_nhwc(outs)
-            if layout == "NCHW4c":
+        if groups > 1:
+            if layout == "NCHW":
```

Comment:
`layout in ['NCHW', 'NCHW4c']`
group conv on cuda has NCHW4c layout

Reply:
I have been working with the CPU/LLVM backend, would that work with the `NCHW4c` backend too?   Conversely, do you expect the CUDA handler throw a reasonable error if `NCHW` is chosen instead of 'NCHW4c'?

Reply:
The default schedule in LLVM backend is `NCHW`, so `NCHW4c` won't work for LLVM unless you add a schedule for that. CUDA has `NCHW4c` but no default layout in`NCHW`. It's reasonable to throw an error in CUDA handler.